src/pages/previous.js

```diff
@@ -0,0 +1,152 @@
+import NavBar from "../components/navBar";
+import Chevron from "../assets/chevron_down.svg";
+import { GET } from "../utils/fetch";
+import { useEffect, useState } from "react";
+
+async function getQuestions() {
+  const weekly_questions = await GET("weekly_questions");
+  const questions = [];
+  for (const value in weekly_questions) {
+    questions.push({
+      id: weekly_questions[value].question_id,
+      body: weekly_questions[value].question.body,
+      time: new Date(weekly_questions[value].week),
+    });
+  }
+  return questions;
+}
+
+const options = {
+  month: "long",
+};
+
+function Previous() {
+  const [questions, setQuestions] = useState([]);
+  const [answers, setAnswers] = useState({});
+  const [answersVisibility, setAnswersVisibility] = useState({});
+
+  useEffect(() => {
+    getQuestions().then((response) => {
+      setQuestions(response);
+    });
+  }, []);
+
+  const viewAnswers = (questionId) => {
+    setAnswersVisibility((prevVisibility) => ({
+      ...prevVisibility,
+      [questionId]: !prevVisibility[questionId],
+    }));
+
+    getAll(questionId);
+  };
+
+  async function getAll(questionId) {
+    const answers = await GET("questions/" + questionId + "/answers");
+    await Promise.all(
+      answers.map((answer) => {
```

Comment:
Ideally we wouldn't have N+1 requests. What are you trying to achieve here? How can we change the API so that this can be done in less requests?

Reply:
I could have a state that verifies if the answers to a certain question have already been fetched. I guess there could be a way to do with a single request but it would be a pretty complex JSON, with the question then answers and in that answers there's the email to the user and so on

Reply:
You could have a JSON in this format:

```json
[
  {
    "id" => 1
    "body" => "..."
    "answers" => [...]
  },
  ...  
]
```

The JSON:API spec handles situations when this could lead to a lot of the answer information being passed around, but in this case each answer will be unique to the question anyway, so that's not even an issue.

Reply:
Yea just changed it and now it does with one request!